### ISSUE TITLE
multiple accounts

### ISSUE THREAD

**Issue by bbigras** [vector-im/riotX-android#26]
Any plans to support multiple accounts?

**Comment by mguentner** [comment]
It would be nice to have this in the reference client from the beginning.

There are a lot of people who have multiple accounts for different roles they fulfill in their lives (i.e. work, private, community, this one software project, the other hardware project etc.) - this calls for a quick way to switch roles during their day.

**Comment by thomas-profitt** [comment]
I want to voice my support for this too:

Multiple accounts is something that needs to be facilitated early in the architecting of an application.

I currently have work and personal Matrix accounts, and use both miniVector-fdroid and Riot.im-fdroid, respectively, on the same device to facilitate it. It's inefficient because they both need to wake to sync and send notifications independently of one-another, and if I want to configure something for both of them, I have to do it twice.

**Comment by bbigras** [comment]
In my case I use a work account since I don't want to log in with my personal account at work and allow my employer to see my personal conversations.

At the same time I would like not having to use my phone to ask a question in #matrix:matrix.org

**Comment by PhieF** [comment]
Yes it would also be really important for me ! :)

**Comment by vc7ee** [comment]
@PhieF please delete your comment and add a +1 reaction to the original comment.

**Comment by laszlovl** [comment]
I would like to join the plea for adding multiple account support to riotX from the start, I hope you will consider my perspective worthwhile enough to keep it as a comment instead of a +1.

What's the problem with all messengers that riotX will be competing with, such as Whatsapp and Signal? They only allow for a single account ("identity") and even worse, they force you to tie that identity to a cellphone number making it almost impossible to have two separate identities such as "work" and "personal".

What's an attribute that almost every other communication ecosystem that predates Whatsapp and Matrix has in common? That support for multiple identities is the norm. You won't be able to find an email client where it's impossible to add more than one account. All XMPP clients out there have supported multiple accounts from the start. IRC users are used to having a dozen of different accounts (networks).

Even the vast majority of modern cellphones has support for dual-SIM, because there's a huge demand from users that want to keep their work identity (phone number) separate from their personal one.

Yet unfortunately all reference clients for Matrix currently lack multi-account support. This makes it a dealbreaker for me and many people I know. Supporting this would make Riot/Matrix really stand out from its competitors on a functional level instead of merely an ideological one.

**Comment by 532910** [comment]
Is any update there?
I still remember that it was in a roadmap for riot-android on play.google.com.

**Comment by ManDay** [comment]
I would like to input my opinion on this matter, although I know that some people will not appreciate it.

I do not oppose multiple accounts as a feature as long as it doesn't get in the way of those who don't use it, but I think it should be much lower on the priority list than certain other features which receive comparitively less attention. I see how it can be useful in many cases, however, for the reasons that I've been told, it seems to be more like a hack around a gone-wrong abuse of Matrix' specification, where instead of one identity in a federation, a user wants to occupy multiple identities simulatenously (NB: I'm neither a Matrix nor a Riot developer and this is solely my interpretation of Matrix and how it should be used). It therefore shouldn't be a "first class citizen" among Riot's features and therefore implemented after Riot is more complete regarding the more essential features.

I'm saying this because I do think that multiple accounts **will** inevitably have an influence on the UI/UX for everyone (apart from binding development resources, which, however, is none of my business to be interested in), unless, of course, they are implemented such that they effectively behave as if multiple instances of Riot were running separately, one for each account.

That said, I understand the desire to have multiple accounts but I fail to understand its urgency for the reasons that @laszlovl for example has mentioned.

For instance, you are speaking of separating personal and work identities, and given that you seem not to be satisfied by using different *Display Names* in rooms which are related to work/personal live respectively (which, personally I think is the way to go down), I assume your goal is to entirely conceal your identity to the point where even further inspection (of, say, the MXID, the IP, etc.) would not yield any indication as to "who is who". Meanwhile, you cite IRC as an example where multiple identities exist, yet

- rarely, if ever, have I met someone who uses IRC privately **and** professionally **and** both on the same device **and** is so concerned about people being able to identify them *while* they are connected from the same device. To be blunt, I would therefore be surprised if your need for this feature in Matrix is actually genuine.
 - Freenode and other networks may provide Cloaks, however and in general, I don't think IRC is designed around anonymity of clients. Even in Matrix, there will be side channels like the read receipt, so you will never get total anonymity and multiple accounts are only obfuscation, so we might as well stay with different display names.

In my opinion, multiple accounts could be realized simply by parallel instances of Riot and therefore without any effect on UI/UX of single-account instances.

**Comment by jtl999** [comment]
> That said, I understand the desire to have multiple accounts but I fail to understand its urgency for the reasons that @laszlovl for example has mentioned.

> For instance, you are speaking of separating personal and work identities, and given that you seem not to be satisfied by using different Display Names in rooms which are related to work/personal live respectively (which, personally I think is the way to go down), I assume your goal is to entirely conceal your identity to the point where even further inspection (of, say, the MXID, the IP, etc.) would not yield any indication as to "who is who". Meanwhile, you cite IRC as an example where multiple identities exist, yet

Many email clients (Outlook, Thunderbird, et al) have supported managing multiple email accounts for years. Being able to utilize multiple identities has been a common usecase for this feature. Then again I'm not going to compare common email protocols versus the nuances of the Matrix protocol and potential side channel leakage.

**Comment by laszlovl** [comment]
> For instance, you are speaking of separating personal and work identities, and given that you seem not to be satisfied by using different Display Names in rooms which are related to work/personal live respectively (which, personally I think is the way to go down)

Just a couple of very common usecases for (work vs. personal) email:

* On my business computer I only want to receive work messages, on my private computer only personal messages, on my phone I want to get both
* When I'm on vacation, I want to disable/log out of my work identity so I don't receive any of those messages, but continue to receive personal messages
* My employer wants to set strict limits on my work identity (like forcing message retention, limiting the domains that can be federated with, password/2fa policies, etc) but I don't want these limits on my personal account
* When I quit my job, my employer wants to assume my work address, but I obviously don't want them to be able to touch my personal account
* I want to keep my personal and work identities separate; if someone googles my work address I don't want them to find references to what I did in my own time

All of these are only possible with multiple account support. 

> multiple accounts could be realized simply by parallel instances of Riot

This isn't possible on Android, and even if it were the UX for that would be much detrimental versus combining multiple accounts in a single interface (like all Email/IRC/XMPP/etc clients do)

**Comment by 532910** [comment]
Shelter can be used as a solutios for those who need one more account right now.
https://github.com/PeterCxy/Shelter
https://f-droid.org/packages/net.typeblog.shelter
https://play.google.com/store/apps/details?id=net.typeblog.shelter

**Comment by bbigras** [comment]
There's Island too.

But I need 3 RiotX right now: 1 for work, 1 for personal and one to play with the p2p demo. Life is not fair.

**Comment by ralaud** [comment]
+1

**Comment by pnsbttl** [comment]
+1

**Comment by trymeouteh** [comment]
This is needed badly.

**Comment by IntelligentesTierMaulApollo13** [comment]
I suppose there is no point in trying to start crowd funding this feature, right? (New Vector recently received 8.5M $ of venture capital)

**Comment by ShadowJonathan** [comment]
I think the developers' attention and priorities lie somewhere else than this feature, but at least an acknowledgement that its on the backlog (or similar) would be nice, @bmarty

**Comment by tgr** [comment]
This a crippling omission for anyone who needs to interact with a non-federated Matrix server (e.g. for work) - there will be no official Matrix phone client available for them, basically.

**Comment by ptman** [comment]
@tgr I'm guessing you mean something by new vector / element with official. But there are other clients. E.g. SchildiChat is a fork of element for android, so you can have two elements at the same time. And there are lots of more android clients: https://matrix.org/clients-matrix . I'm not claiming that it's perfect, but it may help in your situation.

**Comment by tgr** [comment]
@ptman it's very far from perfect. The official clients some with all kinds of assurances (New Vector is a professional company known to do high-quality work; they will likely be around for support as long as Matrix itself is supported; they stand to lose a lot from security issues; etc) that any random client doesn't. Plus people would have to learn two different clients then.

Elsewhere [App Cloner](https://appcloner.app/) was brought up as an option, that sounds less bad.

**Comment by fjrider** [comment]
> Any plans to support multiple accounts?
> 
> **Edit by Benoit**
> 
> The SDK already support multi-account.
> 
> Blocked by: [matrix-org/sygnal#70](https://github.com/matrix-org/sygnal/issues/70)
> 
> Open points:
> 
>     * [ ]  UI/UX
> 
>     * [ ]  What happen in case of soft logout
> 
>     * [ ]  Notifications
> 
>     * [ ]  Logout
> 
>     * [ ]  Login when already connected to another one
> 
>     * [ ]  Create a new account when already connected to another one
> 
>     * [ ]  To be completed

Love the app.  I would like to see multi user support also!

**Comment by uha4** [comment]
Beeing able to have multiple accounts for different purposes is one of my main reason i wanted to switch to matrix and promote it among my friends. This was the reason why i considered matrix better in privacy than e.g. Signal for my usecases.
It is quite disappointing to learn that this is not supported out of the box by most standard clients.

**Comment by Xyz00777** [comment]
What's the state of these feature request? Its one of the oldest request... (im not a programmer) but it cant be so difficult... or?

**Comment by trymeouteh** [comment]
> Beeing able to have multiple accounts for different purposes is one of my main reason i wanted to switch to matrix and promote it among my friends. This was the reason why i considered matrix better in privacy than e.g. Signal for my usecases.
> It is quite disappointing to learn that this is not supported out of the box by most standard clients.

Once this feature is available, I do think Matrix will explode.

**Comment by ShadowJonathan** [comment]
> Once this feature is available, I do think Matrix will explode.

I don't think its necessarily a killer feature, but it's definitely requested from a user-base of power users who'd want to switch between work and personal accounts (and the like)

I think https://github.com/matrix-org/matrix-doc/issues/915 is more related to this, with it's prospect that it enables users to give less thought to where they sign up in the first place, and being able to move their accounts' "home" wherever they like at any time, which gives them more freedom.

**Comment by JokerGermany** [comment]
Because of the bad ios support for xmpp I was thinking of changing to matrix. Still no support for mulitple Accounts oO

I am really glad that i decided to use xmpp a couple of years ago because this feature was missing...

**Comment by btittelbach** [comment]
> I think [matrix-org/matrix-doc#915](https://github.com/matrix-org/matrix-doc/issues/915) is more related to this, with it's prospect that it enables users to give less thought to where they sign up in the first place, and being able to move their accounts' "home" wherever they like at any time, which gives them more freedom.

I don't see how this would solve the problem of needing to interface with several accounts in parallel.

The fact that we don't have UI support for this, is currently preventing me from introducing Matrix into our company, since people don't want to give up their personal accounts _and_ don't want to user their private identiy for work.

**Comment by nkosi23** [comment]
It would be great to have official feedback on the roadmap and/or timeline for this feature, at least so we know whether or not we can expect it soon.

**Comment by SimonBrandner** [comment]
[I believe this is planned](https://github.com/vector-im/roadmap/projects/1#card-48805101)

**Comment by nkosi23** [comment]
To add to the discussion, once thing that hasn't been mentioned/noticed so far is that this feature would likely open up a powerful revenue platform for the project. From what I gather, Element is making money by having users create an account on a premium EMS homeserver. However with the current situation, this is a hit-or-miss: if the user creates an account on a different homeserver the first time he installs Element, the lifetime value of this customer is radically reduced. It will be very difficult to have the individual switch.

However, with multi-account support, Element would be in a position to convince users to create a premium account (as an extra account) any time. EMS accounts, could even include features allowing people to manage multi-accounts more easily (ability to backup conversations from other homeservers, advanced integration of this scenario with their bots and bridges etc...).

**Comment by MWT** [comment]
> [I believe this is planned](https://github.com/vector-im/roadmap/projects/1#card-48805101)

Yes. It's on next right now, so I guess it should be upcoming.

**Comment by Mageek627** [comment]
> Elsewhere [App Cloner](https://appcloner.app/) was brought up as an option, that sounds less bad.

@tgr You got my hopes up for a few seconds, until I realized it's closed source...

**Comment by pnsbttl** [comment]
Otherwise, I have both element & fluffychat installed on the phone...

**Comment by btittelbach** [comment]
> Otherwise, I have both element & fluffychat installed on the phone...

I tried that too, but fluffychat still has some ways to go, though the progress is very promising ;-)
Just one of the problems is the jitsi incompatibilty. fluffychat users can't call element users, etc.

**Comment by a000a** [comment]
I wrote a small script my server executes every few minutes to see if s.o. invited to my other account and invites my primary user to this chat. works only if the homeserver accepts other ones.
https://github.com/turt2live/matrix-wishlist/issues/56#issuecomment-831982522

**Comment by Porkepix** [comment]
> What I don't want / need:
> 
>     1. Decide in each chat with which account I want to send the message. My private account should never (accidentally) try to send a message into a work group.

It's also possible that any message you write is with the account you joined the chan, what many XMPP client does. As for how it's shown, considering this option makes it a non-issue to have chans mixed between accounts as the user is always going to write with the correct one.

It could also be that every chan/user we talk with is a sublist from a section in the list.

Because what you describe really makes me think to what Slack does for different server, but it would sound to me like an half-implementation, and what can makes sense for Slack because of how it works (no decentralization, even if they have some kind of interconnections, the basic idea is still that a server = an account that connects to the chans of this server) doesn't necessarily makes sense for other solutions such as Matrix.

**Comment by flintstone1409** [comment]
> p. s. In principle I would like to duplicate the whole app for each account. An intermediate solution might be to simply publish a green, blue, red and purple version of the Element app in the Google Playstore and on F-Droid, to have up to four Element apps installed simultaneously.

This. At the moment I'm doing basically that by using a debug version as second instance, even with my work instance using dark theme and the private one using the light theme (don't blame me for that, I think the app actually looks better in light mode :P) to instantly see which instance I'm using. This is by far not optimal (because of slightly different feature sets, maybe bugs (which is why I'm using the debug version for my private account), having to do updates to the debug version manually and probably many more), but the best solution for now.

In the end, I really like the idea of just having multiple instances with different colors, because the most important thing is easy and fast switching between accounts and seeing which one I'm using right now without much thinking about it.

**Comment by derNarr** [comment]
The more I think about it, the more I would like to simply have four different coloured versions of Element in the f-droid store that can be installed side bide side. This really solves the same purpose as the ``--profile`` option for the Desktop version of Element. Anyone aware who is in charge for publishing in f-droid?

**Comment by MurzNN** [comment]
I used AppCloner app to clone main Element app with other name, and also can re-color the icon.

**Comment by maste9** [comment]
> The more I think about it, the more I would like to simply have four different coloured versions of Element in the f-droid store that can be installed side bide side. This really solves the same purpose as the `--profile` option for the Desktop version of Element. Anyone aware who is in charge for publishing in f-droid?


As a quickfix: 

Are you aware of https://schildi.chat? It's a fork of element with just some minor style-differences but uses the current versions of element. You'll find it at F-droid so that you could at least use element and schildi side by side on android. I.e. if you need to divide only to contexts from each other.

**Comment by JokerGermany** [comment]
Would love to see this.
At the moment we make emergency plans.
What happens if your complete infrastructure isn't available anymore?

In this case a fallback to another Matrix Server would be great...

**Comment by OtherSystems** [comment]
This feature would be crucial for us to be able to adopt matrix within the organization. We want to run our own node/instance/synapse for data control. But quite some users already have a matrix account, as they would need also an organization account (we want to control federation of certain content) they can't use their mobile for the organizations account. We can't really sell having to use a custom/other client either. 

Is there any hope for this feature in the near future? Any progress updates?

**Comment by maurerbot** [comment]
The only thing stopping me from using matrix as my primary is multiple accounts

**Comment by marceltransier** [comment]
> The only thing stopping me from using matrix as my primary chat is multiple accounts

@maurerbot fluffychat already supports multiple accounts. so now nothing is stopping you ;)

but i also agree that this is a very useful feature! especially if you work/tinker with matrix servers.

**Comment by maurerbot** [comment]
> > The only thing stopping me from using matrix as my primary chat is multiple accounts
> 
> @maurerbot fluffychat already supports multiple accounts. so now nothing is stopping you ;)
> 
> but i also agree that this is a very useful feature! especially if you work/tinker with matrix servers.

Only for matrix.org accounts

**Comment by marceltransier** [comment]
> > > The only thing stopping me from using matrix as my primary chat is multiple accounts
> > 
> > 
> > @maurerbot fluffychat already supports multiple accounts. so now nothing is stopping you ;)
> > but i also agree that this is a very useful feature! especially if you work/tinker with matrix servers.
> 
> Only for matrix.org accounts

no. at least not the version i am using. (0.41.1 from fdroid)

**Comment by github-tomster** [comment]
@thosetryingtocode
https://github.com/vector-im/roadmap/issues/11
this is all a joke 🤡

**Comment by DraconicNEO** [comment]
> I used AppCloner app to clone main Element app with other name, and also can re-color the icon.

The biggest issue here is that it essentially requires you to install a whole new version of the app for every new account which will end up taking up a significant amount of space.

**Comment by lindhe** [comment]
Would this enable the "Dual Messenger" functionality in Android (or at least Samsung Galaxy S21)?

**Comment by uranuspucksaxophone** [comment]
+1 for this issue. Good for compartmentalizing.

**Comment by jittygitty** [comment]
@DraconicNEO Is Appcloner app Open-Source? If so where is the source code? I would not recommend to people to use some app that is not even available on Google Play and is closed-source and who knows what it might do (because technically it can), and let it create a new APKs for you with its own signature keys etc. Sounds pretty risky to me and possibly even against Google Play's terms of service which is why I guess they can not deploy the app on Google Play or other official stores?

**Comment by DraconicNEO** [comment]
> @DraconicNEO Is Appcloner app Open-Source? If so where is the source code? I would not recommend to people to use some app that is not even available on Google Play and is closed-source and who knows what it might do (because technically it can), and let it create a new APKs for you with its own signature keys etc. Sounds pretty risky to me and possibly even against Google Play's terms of service which is why I guess they can not deploy the app on Google Play or other official stores?

That's yet another good reason not to use that method. I wasn't recommending it for people, I'm actually against it because it encourages lazy bandaid solutions for things like the multiple accounts issue instead of actually solving the issue.

**Comment by tazihad** [comment]
Telegram has this feature. I hope it comes in this app too.

**Comment by wokawoka** [comment]
+1

**Comment by schrmh** [comment]
I'm kinda a fan of multiple account support but only if I can use all of them at the same time (e.g. account per room setting & asking which account to join the room with). It should not be done like on Discord (you can switch to another account and then have to switch back to see a chat from the other account...)

**Comment by remram44** [comment]
The Discord one would still be an improvement above the current situation. Contrary to Discord, logging in to a Matrix account requires more than your username and password, since you have to verify keys.

**Comment by JokerGermany** [comment]
I think this can be closed as unplanned. 
I don't think element android will get such a complicate feature **when element x android** is developed and **will have this feature**.

**Comment by ShadowJonathan** [comment]
> will have this feature

Where is that detailed?

**Comment by darkdragon-001** [comment]
ElementX issue got closed and the topic got burried again 😢

**Comment by JokerGermany** [comment]
> ElementX issue got closed and the topic got burried again cry

It wasn't closed.

**Comment by michaelblyons** [comment]
> It wasn't closed.

For Android, it certainly was. The highest-voted Element X issue was closed as a duplicate of https://github.com/vector-im/element-meta/discussions/1832 on Meta, with a fraction of the visibility. If you (dear issue subscriber) had previously up-voted the Android Element X issue, please also do so on the Meta one.

**Comment by JokerGermany** [comment]
https://github.com/vector-im/element-x-android/issues/190 isn't closed..

**Comment by michaelblyons** [comment]
What are you talking about? It's both closed and (when there were objections) locked by an admin.

![Screenshot_20230716-131703](https://github.com/vector-im/element-android/assets/2701605/aad34ee0-2340-4bc8-b46e-5b3c770372e5)

**Comment by DraconicNEO** [comment]
> What are you talking about? It's both _closed_ and when there were objections, it was _locked_ to forestall further complaint.
> 
> If you want to move discussion to Meta, that's fine with me. That's a reasonable organizational move. But you either leave the issue open for tracking by people subscribed to it, _or_ you move the popular issue to Meta to keep the interest tagging.
> 
> ![Screenshot_20230716-131703](https://user-images.githubusercontent.com/2701605/253817815-aad34ee0-2340-4bc8-b46e-5b3c770372e5.png)

Starting to get the feeling they don't want to implement this feature, I hope I'm wrong though.

**Comment by rokejulianlockhart** [comment]
https://github.com/orgs/community/discussions/61069#discussion-5411152

**Comment by kuhlmannmarkus** [comment]
+1

**Comment by DraconicNEO** [comment]
> +1

Don't comment just to bump these issues. That might be necessary on repos run by assholes that mark requests as stale and close them automatically, but Element's devs don't do this.

**Comment by kuhlmannmarkus** [comment]
Sorry for just bumping, here is my reasoning:

We are running a Matrix instance for our nursery school where we as parents communicate with each other and share sensitive information. We now faced the challenge that some of the parents were already using element with some other instance (mostly for their jobs). These users had to choose other Matrix clients to participate in our communications that are not as feature complete as element.
It would be nice to be able to use multiple accounts in element to solve this.

**Comment by jittygitty** [comment]
@kuhlmannmarkus Note that I haven't actually had the chance to try/test TWAKE-CHAT matrix client but it claims to have MULTI-ACCOUNT ability https://twake-chat.com  (see https://github.com/linagora/twake-on-matrix  and apparently it is based up on fluffychat https://github.com/krille-chan/fluffychat  )

**Comment by RokeJulianLockhart** [comment]
What [`source.android.com/docs/devices/admin/managed-profiles#implementation-details`][2] [^2] describes mitigates this. However, it would require client preference synchronisation to be useful, and IDK whether that exists (yet). [^1]

[^2]: [`android.stackexchange.com/revisions/258245/4`][3]

[3]: https://android.stackexchange.com/revisions/258245/4#:~:text=Google%20has%20confirmed%20that%20the%20choice%20to%20not%20publicly%20mirror%20it%20is%20deliberate

[^1]: [`element-meta/discussions/2987`][2]

[2]: https://github.com/element-hq/element-meta/discussions/2987#discussion-8980695

[2]: https://source.android.com/docs/devices/admin/managed-profiles#implementation-details

**Comment by michaelblyons** [comment]
In case you missed it (like I did), Element X has an *experimental* and *unsupported* multi-account feature at https://github.com/element-hq/element-x-android/pull/5285. Encouraging sign, though!

<sub>Grumble grumble issue fragmentation grumble.</sub>

**Comment by ShadowJonathan** [comment]
element X is meant to be the replacement for element android, so I'd expect them to focus more on new features over there, but this is still interesting
